### ISSUE TITLE
Support CUDA Quantum as frontend

### ISSUE THREAD

**Issue by natestemen** [unitaryfund/mitiq#2200]
More detailed description to come, but in short we should add a new folder in [`mitiq/interface`](https://github.com/unitaryfund/mitiq/tree/c5afbd4df6850cacc331a1aa1a477265d4bb584a/mitiq/interface) with all the necessary conversion function from CUDA to Cirq. A preliminary search of the CUDA Quantum [documentation](https://nvidia.github.io/cuda-quantum/latest/index.html) do not show a to QASM function, but I do believe it exists. I will update this issue with details when I find it.

**Comment by purva-thakre** [comment]
>A preliminary search of the CUDA Quantum [documentation](https://nvidia.github.io/cuda-quantum/latest/index.html) do not show a to QASM function, but I do believe it exists. I will update this issue with details when I find it.

I think this is what you are looking for: https://github.com/NVIDIA/cuQuantum/blob/2ac5645a56063d4d83b1c82f833ccdaa75b3736f/python/cuquantum/cutensornet/_internal/circuit_converter_utils.py#L28

**Comment by natestemen** [comment]
I might be missing it, but I don't see any way to import/export qasm in that file.

**Comment by purva-thakre** [comment]
Yeah, you're right. I don't know why I thought it did. I quickly glanced through the file before making my previous comment.

**Comment by natestemen** [comment]
From CUDA team:

> We currently have [support for exporting QASM] for C++ but we need a bit more time to support that in Python.

This confirms both what is visible in their documentation and what Purva found above. Until this is implemented on their side, no action needed here. We will wait for CUDA Quantum to support export/import QASM via their Python API.

**Comment by ACE07-Sev** [comment]
I tried to install on windows, but got this error for it.
```
Exception: Your CUDA version (13000) is too new.
```
If you guys can, try to see if you can install it on windows or mac (no linux distros). If it installs, then perhaps we can add it as a frontend. Shouldn't take long.

**Comment by natestemen** [comment]
Not sure what install process you went through, but installing through `uv` on macOS worked for me:

```bash
$ uv pip install cudaq
Resolved 1 package in 210ms
      Built cudaq==0.12.0.post1
Prepared 1 package in 900ms
Installed 1 package in 0.91ms
 + cudaq==0.12.0.post1
```
